### PR TITLE
Make mass replace work with proxy types

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Adding isCompatibleCMSType allows handling of proxy types in mass search functions involving InputTags

#### PR validation:

The unit tests pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/ParameterSet/python/MassReplace.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def doIt(self,pset,base):
                             nrep = n; nrep.moduleLabel = self._paramReplace.moduleLabel
                             if self._verbose:print("Replace %s.%s[%d] %s ==> %s " % (base, name, i, n, nrep))
                             value[i] = nrep
-                elif isinstance(value, cms.InputTag):
+                elif value.isCompatibleCMSType(cms.InputTag):
                     if value == self._paramSearch:
```
Comment (slava77) [inline FWCore/ParameterSet/python/MassReplace.py:42]: I suspect that something like ```eqWproxy``` from https://github.com/cms-sw/cmssw/pull/30071/files#diff-c2356aa3d79bc5ad0d7e89ea86cca856R12-R16 will be needed here, and apparently in the VInputTag case above

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
+1

File: FWCore/ParameterSet/python/MassReplace.py (hunk 1 above)
Comment (slava77) [reply FWCore/ParameterSet/python/MassReplace.py:42]: this ```and value``` is more elegant

**Comment by silviodonato** [comment]
+1

**Comment by silviodonato** [comment]
merge

**Pull request merged**